File: Go/factorialIterative.go
package main

import "fmt"

func factorial(n int) int {
	sum := 1

	for i := 2; i <= n; i++{
		sum*=i
	}
	return sum
}

func main() {
	fmt.Print("Enter a number: ")
	var n int
	fmt.Scanf("%d", &n)

	if n >= 0 {
		fmt.Println("Factorial de", n, "es", factorial(n))
	}else{
		fmt.Println("Choose another number")
	}
}

File: Python/factorialIterative.py
def factorial(n):
    if(n == 0 or n == 1):
        return 1
    fact = 1
    for i in range(2,n+1):
        fact*= i
    return fact
    

def main():
    
    n = int(input("Enter a number: "))
    
    if n >= 0:
        print(f"Factorial: {factorial(n)}")
    else:
        print(f"Choose another number")
  

if __name__ == "__main__":
    main()
